packages/fetchai/connections/soef/connection.py

```diff
@@ -971,9 +982,21 @@ async def _stop_periodic_ping_task(self) -> None:
                 await self._ping_periodic_task
             self._ping_periodic_task = None
 
+    async def _check_server_reachable(self) -> None:
+        """Check network connection is ok."""
+        await asyncio.wait_for(
+            self._request_text(
+                "get", self.base_url, timeout=self.connection_check_timeout
+            ),
+            timeout=self.connection_check_timeout,
+        )
+
     async def connect(self) -> None:
         """Connect channel set queues and executor pool."""
         self._loop = asyncio.get_event_loop()
+
+        await self._check_server_reachable()
```

Comment:
what happens if it fails? Exception raised?

Reply:
yes